Have the tests link against the shared libs instead of static ones.

--- mumps-4.9.2.dfsg/examples/Makefile~	2009-11-05 02:05:21.000000000 -0500
+++ mumps-4.9.2.dfsg/examples/Makefile	2009-12-07 17:54:56.000000000 -0500
@@ -19,28 +19,28 @@
 
 include $(topdir)/Makefile.inc
 
-LIBMUMPS_COMMON = $(libdir)/libmumps_common$(PLAT).a
+LIBMUMPS_COMMON = $(libdir)/libmumps_common$(PLAT).so
 
 
-LIBSMUMPS = $(libdir)/libsmumps$(PLAT).a $(LIBMUMPS_COMMON)
+LIBSMUMPS = $(libdir)/libsmumps$(PLAT).so $(LIBMUMPS_COMMON)
 
 ssimpletest:  $(LIBSMUMPS) $$@.o 
 	$(FL) -o $@ $(OPTL) ssimpletest.o $(LIBSMUMPS)  $(LORDERINGS) $(LIB) $(LIBBLAS) $(LIBOTHERS)
 
 
-LIBDMUMPS = $(libdir)/libdmumps$(PLAT).a $(LIBMUMPS_COMMON)
+LIBDMUMPS = $(libdir)/libdmumps$(PLAT).so $(LIBMUMPS_COMMON)
 
 dsimpletest: $(LIBDMUMPS) $$@.o 
 	$(FL) -o $@ $(OPTL) dsimpletest.o $(LIBDMUMPS)  $(LORDERINGS) $(LIB) $(LIBBLAS) $(LIBOTHERS)
 
 
-LIBCMUMPS = $(libdir)/libcmumps$(PLAT).a $(LIBMUMPS_COMMON)
+LIBCMUMPS = $(libdir)/libcmumps$(PLAT).so $(LIBMUMPS_COMMON)
 
 csimpletest: $(LIBCMUMPS) $$@.o 
 	$(FL) -o $@ $(OPTL) csimpletest.o $(LIBCMUMPS)  $(LORDERINGS) $(LIB) $(LIBBLAS) $(LIBOTHERS)
 
 
-LIBZMUMPS = $(libdir)/libzmumps$(PLAT).a $(LIBMUMPS_COMMON)
+LIBZMUMPS = $(libdir)/libzmumps$(PLAT).so $(LIBMUMPS_COMMON)
 
 zsimpletest: $(LIBZMUMPS) $$@.o 
 	$(FL) -o $@ $(OPTL) zsimpletest.o $(LIBZMUMPS)  $(LORDERINGS) $(LIB) $(LIBBLAS) $(LIBOTHERS)
@@ -57,19 +57,19 @@
 	$(CC) $(OPTC) $(INC) -I. -I$(topdir)/include -c $*.c
 
 
-$(libdir)/libsmumps$(PLAT).a:
+$(libdir)/libsmumps$(PLAT).so:
 	@echo 'Error: you should build the library' $@ 'first'
 	exit -1
 
-$(libdir)/libdmumps$(PLAT).a:
+$(libdir)/libdmumps$(PLAT).so:
 	@echo 'Error: you should build the library' $@ 'first'
 	exit -1
 
-$(libdir)/libcmumps$(PLAT).a:
+$(libdir)/libcmumps$(PLAT).so:
 	@echo 'Error: you should build the library' $@ 'first'
 	exit -1
 
-$(libdir)/libzmumps$(PLAT).a:
+$(libdir)/libzmumps$(PLAT).so:
 	@echo 'Error: you should build the library' $@ 'first'
 	exit -1
 
